usr/src/uts/common/fs/zfs/sys/dbuf.h

```diff
@@ -347,6 +347,12 @@ boolean_t dbuf_is_metadata(dmu_buf_impl_t *db);
 	(dbuf_is_metadata(_db) &&					\
 	((_db)->db_objset->os_secondary_cache == ZFS_CACHE_METADATA)))
```

Comment:
I guess that doesn't really work (it doesn't evaluate to `result`).  You'd have to convert this to an inline function.